### DISCUSSION TITLE
Deployment Problem Using Next js

### DISCUSSION THREAD

**Discussion by SalmanSakri** [vercel/community#5987] [Help] [answered]
### Summary

<html>
<body>

TypeError: fetch failed
18:27:40.462 | at Object.fetch (node:internal/deps/undici/undici:11372:11)
18:27:40.462 | at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
18:27:40.462 | cause: Error: connect ECONNREFUSED 127.0.0.1:3000
18:27:40.462 | at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1555:16)
18:27:40.462 | at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
18:27:40.462 | errno: -111,
18:27:40.462 | code: 'ECONNREFUSED',
18:27:40.462 | syscall: 'connect',
18:27:40.462 | address: '127.0.0.1',
18:27:40.462 | port: 3000
18:27:40.462 | }
18:27:40.462 | }
18:27:40.462 | TypeError: fetch failed
18:27:40.462 | at Object.fetch (node:internal/deps/undici/undici:11372:11)
18:27:40.463 | at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
18:27:40.463 | cause: Error: connect ECONNREFUSED 127.0.0.1:3000
18:27:40.463 | at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1555:16)
18:27:40.463 | at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
18:27:40.463 | errno: -111,
18:27:40.463 | code: 'ECONNREFUSED',
18:27:40.463 | syscall: 'connect',
18:27:40.463 | address: '127.0.0.1',
18:27:40.463 | port: 3000
18:27:40.463 | }
18:27:40.463 | }
18:27:40.463 |  
18:27:40.463 | Error occurred prerendering page "/minar". Read more: https://nextjs.org/docs/messages/prerender-error
18:27:40.463 | TypeError: fetch failed
18:27:40.463 | at Object.fetch (node:internal/deps/undici/undici:11372:11)
18:27:40.463 | at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
18:27:40.581 | ✓ Generating static pages (13/13)
18:27:40.593 |  
18:27:40.594 | > Export encountered errors on following paths:
18:27:40.594 | /minar/page: /minar
18:27:40.669 | Error: Command "npm run build" exited with 1
18:27:40.912

<br class="Apple-interchange-newline">
</body>
</html>

### Steps to Reproduce

error

TypeError: fetch failed
18:27:40.462 | at Object.fetch (node:internal/deps/undici/undici:11372:11)
18:27:40.462 | at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
18:27:40.462 | cause: Error: connect ECONNREFUSED 127.0.0.1:3000
18:27:40.462 | at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1555:16)
18:27:40.462 | at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
18:27:40.462 | errno: -111,
18:27:40.462 | code: 'ECONNREFUSED',
18:27:40.462 | syscall: 'connect',
18:27:40.462 | address: '127.0.0.1',
18:27:40.462 | port: 3000
18:27:40.462 | }
18:27:40.462 | }
18:27:40.462 | TypeError: fetch failed
18:27:40.462 | at Object.fetch (node:internal/deps/undici/undici:11372:11)
18:27:40.463 | at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
18:27:40.463 | cause: Error: connect ECONNREFUSED 127.0.0.1:3000
18:27:40.463 | at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1555:16)
18:27:40.463 | at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
18:27:40.463 | errno: -111,
18:27:40.463 | code: 'ECONNREFUSED',
18:27:40.463 | syscall: 'connect',
18:27:40.463 | address: '127.0.0.1',
18:27:40.463 | port: 3000
18:27:40.463 | }
18:27:40.463 | }
18:27:40.463 |  
18:27:40.463 | Error occurred prerendering page "/minar". Read more: https://nextjs.org/docs/messages/prerender-error
18:27:40.463 | TypeError: fetch failed
18:27:40.463 | at Object.fetch (node:internal/deps/undici/undici:11372:11)
18:27:40.463 | at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
18:27:40.581 | ✓ Generating static pages (13/13)
18:27:40.593 |  
18:27:40.594 | > Export encountered errors on following paths:
18:27:40.594 | /minar/page: /minar
18:27:40.669 | Error: Command "npm run build" exited with 1
18:27:40.912

Code 

import Image from 'next/image'
import Navbar from "../Navbar/navbar"
import Footer from "../Footer/footer"
import Header from '../Header/header';
import Loading from '../Loading/loading';
import "../Loading/loading.css"
import "./minar.css"
import Link from 'next/link';
import loading from '../Loading/loading';
import Form from "../component/form"
// FETCH DATA WITH AN API
const getData = async () => {
  const res = await fetch(`${process.env.BASE_API_URL}/api/product`);

  if (!res.ok) {
    throw new Error("Something went wrong");
  }

  return res.json();
};

const MinarPhoto = async () => {
  if(!process.env.BASE_API_URL){
    return (
      <>
      <h1>hii</h1>
      </>
    );
  }
  const posts = await getData();
  return (
    <section>
      <Header />
      <Navbar />
      <div className="selected-container">
        {
          posts.data.minar.map((value) => (
            <div className="selected-grid" key={value.id}>
              <Link href={`/minar/${value.id}`}>
                <div className="selected-product">
                  <Image src={value.imgUrl} loading="lazy"
                    width={400}
                    height={400}s
                    alt="Photo by Austin Wade" />

                  <div className="selected-product-info">
                    <div>
                      <Link href={`/minar/${value.id}`} className="selected-product-title">{value.title}</Link>
                    </div>
                  </div>

                </div>
              </Link>
            </div>
          ))
        }

      </div>
      <Form />
      <Footer />
    </section>
  )
}

export default MinarPhoto

**Comment by swarnava** [comment]
Does your `BASE_API_URL` point to localhost `127.0.0.1:3000` ? Sounds like that's the issue then. 

```
18:27:40.462 | TypeError: fetch failed
18:27:40.462 | at Object.fetch (node:internal/deps/undici/undici:11372:11)
18:27:40.463 | at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
18:27:40.463 | cause: Error: connect ECONNREFUSED 127.0.0.1:3000
18:27:40.463 | at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1555:16)
18:27:40.463 | at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) 
```

**Reply by SalmanSakri** [reply]
I point to this  BASE_API_URL=http://localhost:3000

**Reply by SalmanSakri** [reply]
api/product route.js code below

import data from "../../jsondata.json";
import { NextApiRequest, NextApiResponse } from 'next';
// console.log(data);
export async function GET () {
   if(!data){ 
      return NextResponse.json({
         massage:"no data found"
      })
   }
   return NextResponse.json({
    data,
   })
  };

**Reply by swarnava** [reply, marked as answer]
Yeah you need to replace with an actual URL when you are deploying at Vercel. `localhost` will not work. You can rely on System variable `VERCEL_URL` to get the URL on demand: https://vercel.com/docs/projects/environment-variables/system-environment-variables#system-environment-variables
